apps/indexer/src/api/mappers/transactions.ts

```diff
@@ -0,0 +1,153 @@
+import { z } from "@hono/zod-openapi";
+import { transaction, transfer, delegation } from "ponder:schema";
+import { isAddress } from "viem";
+
+export type DBTransaction = typeof transaction.$inferSelect & {
+  delegations: DBDelegation[];
+  transfers: DBTransfer[];
+};
+export type DBTransfer = typeof transfer.$inferSelect;
+export type DBDelegation = typeof delegation.$inferSelect;
+
+export const AffectedSupplyEnum = z.enum(["CEX", "DEX", "LENDING", "TOTAL"]);
+
+export type AffectedSupply = z.infer<typeof AffectedSupplyEnum>;
+
+export const TransactionsRequestSchema = z.object({
+  limit: z.coerce
+    .number()
+    .int()
+    .min(1, "Limit must be a positive integer")
+    .max(100, "Limit cannot exceed 100")
+    .default(50)
+    .optional(),
+  offset: z.coerce
+    .number()
+    .int()
+    .min(0, "Offset must be a non-negative integer")
+    .default(0)
+    .optional(),
+  sortBy: z.enum(["timestamp"]).default("timestamp").optional(),
+  sortOrder: z.enum(["asc", "desc"]).default("desc").optional(),
+  from: z
+    .string()
+    .refine((addr) => addr && isAddress(addr))
+    .optional(),
+  to: z
+    .string()
+    .refine((addr) => addr && isAddress(addr))
+    .optional(),
+  minAmount: z.coerce.number().optional(),
+  maxAmount: z.coerce.number().optional(),
+  affectedSupply: z.array(AffectedSupplyEnum).optional(),
```

Comment:
could we avoid using enum by having the actual strings? I ask this because the codegen doesn't handle enum on path in a good way